src/chain/chain.rs

```diff
@@ -537,24 +536,17 @@ impl Chain {
             .should_vote_for_merge(self.elder_size(), self.neighbour_infos())
     }
 
-    /// Finalises a split or merge - creates a `GenesisPfxInfo` for the new graph and returns the
-    /// cached and currently accumulated events.
-    pub fn finalise_prefix_change(&mut self) -> Result<PrefixChangeOutcome, RoutingError> {
-        // TODO: Bring back using their_knowledge to clean_older section in our_infos
-        self.check_and_clean_neighbour_infos(None);
-        self.state.change = PrefixChange::None;
-
+    /// Gets the data needed to initialise a new Parsec instance
+    pub fn get_parsec_reset_data(&mut self) -> Result<ParsecResetData, RoutingError> {
         let remaining = self.chain_accumulator.reset_accumulator(&self.our_id);
         let event_cache = mem::replace(&mut self.event_cache, Default::default());
         let merges = mem::replace(&mut self.state.merging, Default::default())
             .into_iter()
             .map(|digest| AccumulatingEvent::NeighbourMerge(digest).into_network_event());
-        self.state.handled_genesis_event = false;
 
-        info!("{} - finalise_prefix_change: {:?}", self, self.our_prefix());
-        trace!("{} - finalise_prefix_change state: {:?}", self, self.state);
+        self.state.handled_genesis_event = false;
```

Comment:
Nitpick, but it's weird a method called `get_parsec_reset_data` is actually mutating something. Maybe rename it to `prepare_parsec_reset`, `begin_parsec_reset` or something like that?